### DISCUSSION TITLE
Collect ideas on new features and bug fixes for CarlaViz 0.9.14

### DISCUSSION THREAD

**Discussion by mjxu96** [mjxu96/carlaviz#60] [General]
Hey everyone,

I've been preparing for a new release for CarlaViz recently. My current ideas for this new release includes:
- Features
  - Support for Carla 0.9.14.
  - Map change during the simulation.
  - Crossloads/stop signs loading during initialization to reduce real-time traffic.
  - Brand new XVIZ cpp library with complete protobuf protocol support.
  - Update the code to allow support for other simulators.
  - Better CI/CD and docker deployment.
- Bug fixes:
  - Fix subscription/un-subscription bug when the backend receives unknown sensors.
  - Re-structure carla backend for better processing speed.

Please leave comments here if you want any new features or notice any bugs that need my attention. Thanks.

Best,
Minjun

**Comment by ksuszka** [comment]
We use CarlaViz to attach to our simulation environment to run a suite of simulation tests. Due to various leaks in Carla we sometimes restart Carla docker to have "fresh" Carla. Unfortunately, at the moment we need to restart CarlaViz after such restart as it is not able to reconnect to new Carla instance by itself. It would be great if it could reconnect to new Carla instance listening on the same addres/port by itself.

**Reply by mjxu96** [reply]
I've implemented this feature in https://github.com/mjxu96/carlaviz/pull/62

Thanks for the suggestion!

**Comment by mrtuwu** [comment]
Hi! I'd be happy to have the ability to see the scene in the camera window from a different angle and switch it while the test is running the same way I'm able to change the RViz view mode (top-down, perspective, driver).

**Reply by mjxu96** [reply]
This can be done with the current version as long as you attach cameras to the position you mentioned. 

Let me know if there are any other concerns.

**Comment by AlanHuangT** [comment]
CarlaViz looks like a mini version of foxglove(https://foxglove.dev/).  Any plan to compare these two visualizer ?

**Reply by mjxu96** [reply]
I just took a look at their [Carla visualization tool](https://github.com/collabora/carlafox). It looks awesome and seems to be able to change the layout in the web visualization.

I don't think CarlaViz, a free open source project developed during my leisure time, can be compared or can be as powerful as this tool developed by an entire commercialized company. I just want to code something when I am out of work and try something interesting. 

Another aspect is that, in CarlaViz, I am more interested in the backend cpp part and try to play with the latest standard in this project instead of frontend part. Currently the frontend of CarlaViz is __just__ enough for visualization.

**Comment by mjxu96** [comment]
Finally I've implemented all features and fixed all bugs in this post in https://github.com/mjxu96/carlaviz/pull/62.

New changes have been merged into master.
